0001-customer-onboarding.md

```diff
@@ -0,0 +1,122 @@
+- Community Issue: [scipian/community#6](https://github.com/scipian/community/issues/6)
+
+# Summary
+[summary]: #summary
+
+Scipian customer onboarding will allow a seamless process for customers to
+onboard to the Scipian platform. At a high level, this includes creation of
+namespaces, RBAC rules, and importing Terraform state from a location external
+to Scipian.
+
+# Motivation
+[motivation]: #motivation
+
+There is currently process for a new customer to be onboarded to Scipian. Having
+a smooth process in place will be paramount to onboarding multiple customers
+who want to use Scipian. Having a low barrier of entry will encourage
+teams to use Scipian.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Scipian builds on and uses many Kubernetes primitives. Some of these include 
+namespaces and RBAC rules. This allows multiple teams to use Scipian, while 
+keeping each team's data and interactions with Scipian separated from eachother.
+
+For onboarding a new team to the Scipian platform, the following is created:
+
+- **Namspace**. A new namespace in the Scipian cluster will be created for a
+given team, under a unique name tied to a specific project or application.
+- **Role**. This will define a role in Scipian as to what operations are allowed 
+in the cluster for anyone asigned to that role. It will be tied to the created 
+namespace, and only allow the least privileges required to create Workspaces and 
+Runs in Scipian.
+- **RoleBinding**. This will bind the created role to a set of users associated 
+with the onboarding team.
+
+If a team is already using Terraform to manage their infrastructure, Scipian
+supports the importing of Terraform state from a location external to Scipian.
+Scipian will pull this state and push it to it's own managed backend.
+
+Below is the flow when onboarding a new customer to Scipian:
+
+- New customer requests to be onboarded to Scipian
+- Customer provides list of users, (the email associated with the IDP Scipian 
+uses) of all the members who will need to use Scipian
+- A Scipian admin will create the Namespace and RBAC roles for the onboarding 
+team
+- Onboarding team members will use the Scipian Authentication CLI to set up 
+their kubeconfigs for interfacing Scipian.
+- The onboarding team will create a Kubernetes secret in their namespace with 
+the AWS credentials needed for Scipian to pull Terraform state from an S3 bucket
+- Scipian will pull that state into it's own backend
+- Customer will then be onboarded and ready to use the Scipian platform
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+- RBAC rules and Namespace will be created manually by a Scipian admin using
+yaml and `kubectl`. This could be automated later via an onboarding UI, but that
+is beyond the scope of this RFC.
+- Terraform state "puller" will be a Kubernetes job that runs a 
```

Comment:
This section is part that I'm wondering if it should be it's own RFC. 

Does this section need to include parts about how the onboarding team will need to provide their own docker container with the appropriate secret and terraform information inside in order to interact with their imported terraform?

We specified `scipctl` here, but will this import be limited to infrastructure created by `scipctl` and `stack` or will it be open to any terraform infrastructure? What would supporting any terraform managed infrastructure mean for mapping the current Terraform state workspace structure to the Scipian expected structure?

Reply:
I think that there's a line somewhere between getting Scipian ready for a new customer to use (RBAC, importing existing infra) and actually using Scipian. The latter will depend on good documentation, the former are steps *we* as Scipian developers/administrators need to have in place for a new customer.

I would have to check a version, but I believe we can take any version of Terraform after they introduced `workspaces` and `runs`. This should probably be documented as part of the using scipian documentation.

It should also be noted that this RFC is to support the onboarding of only a handful of internal customers we have right now. I expect more RFC's to follow as we iterate on this feature of Scipian in the future.

Reply:
Specifically, this first RFC and onboarding process will focus on customers already using `stack`, as that is the immediate need right now.